Modules/_sqlite/statement.c

```diff
@@ -48,29 +48,37 @@ typedef enum {
     TYPE_UNKNOWN
 } parameter_type;
 
-int pysqlite_statement_create(pysqlite_Statement* self, pysqlite_Connection* connection, PyObject* sql)
+pysqlite_Statement *
+pysqlite_statement_create(pysqlite_Connection *connection, PyObject *sql)
 {
     const char* tail;
     int rc;
     const char* sql_cstr;
     Py_ssize_t sql_cstr_len;
     const char* p;
 
-    self->st = NULL;
-    self->in_use = 0;
-
     assert(PyUnicode_Check(sql));
 
     sql_cstr = PyUnicode_AsUTF8AndSize(sql, &sql_cstr_len);
     if (sql_cstr == NULL) {
-        rc = PYSQLITE_SQL_WRONG_TYPE;
-        return rc;
+        PyErr_SetString(pysqlite_Warning,
+                        "SQL is of wrong type. Must be string.");
```

Comment:
You may add Py_TYPE(sql)->tp_name in the error message.

```suggestion
                        "SQL is of wrong type ('%s'). Must be string.");
```

Reply:
BTW; this should probably have been a `TypeError`, not a `Warning`.